File: integration_tests/rap/src/main/java/org/robolectric/rap/ExtendedShadowApplication.java
package org.robolectric.rap;

import static android.os.Build.VERSION_CODES.P;
import static android.os.Build.VERSION_CODES.S;

import android.app.Application;
import org.robolectric.annotation.Implements;
import org.robolectric.annotation.Resetter;
import org.robolectric.shadows.ShadowApplication;

/** This shadow tests the logic that emits resetters in RAP when a min/max sdk is specified */
@Implements(value = Application.class, minSdk = P, maxSdk = S)
public class ExtendedShadowApplication extends ShadowApplication {

  @Resetter
  public static void reset() {}
}

File: integration_tests/rap-ksp/src/main/kotlin/org/robolectric/rap/ksp/ExtendedShadowApplication.kt
package org.robolectric.rap.ksp

import android.app.Application
import android.os.Build.VERSION_CODES.P
import android.os.Build.VERSION_CODES.S
import org.robolectric.annotation.Implements
import org.robolectric.annotation.Resetter
import org.robolectric.shadows.ShadowApplication

/** This shadow tests the logic that emits resetters in KSP when a min/max sdk is specified. */
@Implements(value = Application::class, minSdk = P, maxSdk = S)
class ExtendedShadowApplication : ShadowApplication() {
  companion object {
    @JvmStatic @Resetter fun reset() {}
  }
}
